### ISSUE TITLE
correct caching setup for full page plugin?

### ISSUE THREAD

**Issue by MattAppleton** [OleVik/grav-plugin-fullpage#2]
Hi there 

Testing out full page with a plain  / new install of full page plugin with GRAV 1.4.0-rc.1...

Not sure why, but if I take this 'plain' install and add the full page plugin, I can build a successful full page page — EXCEPT when I _refresh_ the page I lose the fullpage.js and the full page.css and the page loses most of the full page styling...

Seems to be something to do with the caching... any tips on how to set this up correctly?

thanks

**Comment by MattAppleton** [comment]
Just confirming that if i repeat the test with GRAV release 1.3.10, I get the same result after installing the plugin — first full page I make works great, until I hit refresh, then the .css and .js relevant to full page script evaporate...

**Comment by MattAppleton** [comment]
And even if I ensure the .js and .css files are loaded into the page by adding them to the full page twig template (and I check they are loaded in the browser by inspecting) we still find the page loses all its full page behaviours if we refresh the browser...

**Comment by OleVik** [comment]
Temporarily disabling the standard cache;

```
cache:
  enabled: false
```

works. I am evaluating a fix by elevating the process handling.

**Comment by OleVik** [comment]
Issue is with how Twig-variables are handled; [upstream-issue](https://github.com/getgrav/grav/issues/1865).

**Comment by MattAppleton** [comment]
Thanks for identifying the issue, looking forward to seeing what might be going on there :)

**Comment by MattAppleton** [comment]
Ole — had a brief chat with @rhukster...

in a nutshell

"its just that you can't really do that when caching is on stuff in rendered by twig is cached, and twig is not processed anymore have you tried `never_cache_twig: true` in the page?"

— setting `never_cache_twig: true` works fine, I can confirm... much less heavy-handed than `cache_enabled: false` I imagine

**Comment by MattAppleton** [comment]
Spoke briefly with @rhukster, who suggested setting`never_cache_twig: true` on a per page basis... but that doesn't work...

he mentioned 

"its just that you can't really do that
when caching is on
stuff in rendered by twig is cached, and twig is not processed anymore"

— just wondering if there is a way to use your very useful user/admin functionality as some kind of particle without needing the whole plugin framework...

basically the bit that turns `---` into a full page section!

**Comment by OleVik** [comment]
That is basically the [`buildContent`-method](https://github.com/OleVik/grav-plugin-fullpage/blob/master/Utilities.php#L100) from `Utilities.php`, which you could leverage or adapt. It recursively processes the page-output from Grav, but should be applicable to HTML-structures more broadly.

As for the caching-issue, it's on my todo-list of things to find a solution for, but this may take time.

**Comment by MattAppleton** [comment]
Thanks for the update —  full page working OK in GRAV (so far!) simply creating a fp-section for each image in the page.media... 

I can totally see how your setup would allow more flexible / wide-ranging use of FullPage possibilities within GRAV's admin framework though, definitely still interested in seeing what happens with this plugin.

**Comment by kafkiano** [comment]
Hi 
unfortunately i am experiencing the same issue like above. The workaround with `cache_enable: false` on fullpage.md is working fine for dev. With light contend and less users i assume not getting any problems even in production. 

I see the potential of this plugin, before i worked always with fullpage as static page. So im trying to help to fix the chaching-issue but sadly i didn't understood the moving asset-logic until now, and implementing maybe some page template for gui creation of fullpage.

Thx very much for this plugin

**Comment by OleVik** [comment]
This should be fixed by [v1.2.0](https://github.com/OleVik/grav-plugin-fullpage/releases/tag/v1.2.0), available on the GPM shortly.
